demo/combo-box-basic-demos.html

```diff
@@ -6,91 +6,155 @@
+              output.textContent = event.target.value;
+            });
+          });
+        </script>
+      </template>
+    </vaadin-demo-snippet>
 
-    <h3>Allow Custom Values</h3>
+    <h3>Allow custom values</h3>
     <p>Allow the user to set any value for the field in addition to selecting a value from the dropdown menu.</p>
     <vaadin-demo-snippet id="combo-box-basic-demos-allow-custom-values" when-defined="vaadin-combo-box">
       <template preserve-content>
-        <vaadin-combo-box label="Element" allow-custom-value></vaadin-combo-box>
-        <p>Selected value: <span></span></p>
-
+        <vaadin-combo-box label="Fruit" allow-custom-value></vaadin-combo-box>
+        <p>Selected value: <span></span></span></p>
         <script>
           window.addDemoReadyListener('#combo-box-basic-demos-allow-custom-values', function(document) {
             const comboBox = document.querySelector('vaadin-combo-box');
             const span = document.querySelector('span');
+            const fruits = ['Apple', 'Orange', 'Banana'];
+            comboBox.items = fruits;
+            comboBox.value = span.textContent = 'Apple';
+
+            comboBox.addEventListener('change', function() {
+              span.textContent = comboBox.value;
+            });
+          });
+        </script>
+      </template>
+    </vaadin-demo-snippet>
+
+    <h3>Storing custom values</h3>
+    <vaadin-demo-snippet id="combo-box-second-custom-value" when-defined="vaadin-combo-box">
+      <template preserve-content>
+        <vaadin-combo-box label="Project" allow-custom-value></vaadin-combo-box>
+        <p>Selected id: <span></span></span></p>
+        <script>
+          window.addDemoReadyListener('#combo-box-second-custom-value', function(document) {
+            const comboBox = document.querySelector('vaadin-combo-box');
+            const span = document.querySelector('span');
 
+            comboBox.items = fetchProjects();
+            comboBox.itemValuePath = 'id';
+            comboBox.itemLabelPath = 'name';
+
+            if (comboBox.items.length > 0) {
+              comboBox.value = comboBox.items[0].name;
+              span.textContent = comboBox.items[0].id;
+            }
             comboBox.addEventListener('value-changed', function() {
               span.textContent = comboBox.value;
             });
             comboBox.addEventListener('custom-value-set', function(e) {
-              // Prevents setting the value property automatically.
               e.preventDefault();
 
-              comboBox.value = e.detail;
-              span.textContent = 'custom value "' + e.detail + '" set!';
+              // insertProject calls an Ajax request to insert a project asynchronously
+              insertProject(e.detail, function(newProject) {
+                comboBox.items.push(newProject);
+                comboBox.value = newProject.name;
+                span.textContent = newProject.id + ' set!';
```

Comment:
The value change listener should handle this. The additional " set!" logic should be handled in the value change listener also, but IMO it doesn't bring much extra value and can be left out completely.

Reply:
Done